### ISSUE TITLE
Missing function computeROI in example ximgproc samples disparity_filtering.cpp

### ISSUE THREAD

**Issue by JohnVorwald** [opencv/opencv_contrib#2908]
##### System information (version)

- OpenCV => The sample code [disparity_filterin.cpp](https://github.com/opencv/opencv_contrib/blob/master/modules/ximgproc/samples/disparity_filtering.cpp) has a function call [Rect computeROI(Size2i src_sz, Ptr<StereoMatcher> matcher_instance);](https://github.com/opencv/opencv_contrib/blob/f5d7f6712d4ff229ba4f45cf79dfd11c557d56fd/modules/ximgproc/samples/disparity_filtering.cpp#L14), but no function is provided.  How to define that function?

##### Steps to reproduce

The code is referenced in example [Disparity map post-filtering](https://docs.opencv.org/master/d3/d14/tutorial_ximgproc_disparity_filtering.html).  

This is an issue.  An example is provided without documentation of a function that is used,  function computeROI.
I have checked everywhere I can think to check, for an answer.
The problem exists in the master version.
No additional code sample is needed to describe the problem, the problem is present in the the example code.  Maybe my question is how to create the computeROI function.

##### Issue submission checklist

 - [X ] I report the issue, it's not a question
   
 - [X ] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [X ] I updated to latest OpenCV version and the issue is still there
     
 - [X ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by JohnVorwald** [comment]
I believe this [link](http://amroamroamro.github.io/mexopencv/opencv_contrib/disparity_filtering_demo.html) provides a clue.   Is this the correct function?  Can the correct function be inserted into the sample code, with some 

```
Rect computeROI(Mat left_for_matcher, Ptr<StereoMatcher> matcher)
{
    int xmin = matcher.MinDisparity + matcher.NumDisparities - 1 + matcher.BlockSize/2;
    int xmax = size(left_for_matcher,2) + matcher.MinDisparity - matcher.BlockSize/2;
    int ymin = matcher.BlockSize/2;
    int ymax = size(left_for_matcher,1) - matcher.BlockSize/2;
    ROI = [xmin, ymin, xmax - xmin, ymax - ymin];
    return Rect(ROI);
}
```

**Comment by berak** [comment]
> but no function is provided. 

[how about this one](https://github.com/opencv/opencv_contrib/blob/a4e601f36d135fdbc5e0ab11bafe8d4f8ba76b39/modules/ximgproc/samples/disparity_filtering.cpp#L461-L477) ? (you only missed it ?)

**Comment by JohnVorwald** [comment]
I believe that's just the function header.

**Comment by alalek** [comment]
Sample `disparity_filtering.cpp` is compiled by [nightly CI builds](http://pullrequest.opencv.org/buildbot/builders/master-contrib-lin64) successfully.

---

Usage questions should go to Users OpenCV Q/A forum: https://forum.opencv.org/

**Comment by JohnVorwald** [comment]
Why was this question closed?  The code compiles, but is missing a function.  The function is not described, so it is not easily implemented.  The question is about compiling the sample code, which does compile, but executing the code, which doesn't execute due to missing function.

**Comment by berak** [comment]
i just checked the sample code,
* it compiles without warnings
* it runs without trouble
* it produces a correctly filtered disparity image

so, @JohnVorwald , if you still think, there is a problem, can you please be more explicit about it ?

>  The code compiles, but is missing a function. 

does not make any sense. any warnings / errors on your side ?

> the question is about executing the code, which doesn't execute due to missing function.

again,  you have some kind of proof, that it doesn't ?

**Comment by JohnVorwald** [comment]
You are correct, I am in error.  The error I get on my side is a link error, missing function.  But, the function starts at line 461.  Somehow, the local copy I'm using is missing that function.  I may have created the local working copy using a screen copy from one file to the next, and only copied down to the end of main, not realizing there was a function below main.  And, my search wasn't thorough enough, I stopped at the first location in the file, corresponding to the function header.  So, the local source file that I used,  which was copied incorrectly from a local clone, dropped the function definition.  I could hunt deeper to explain why I made the error, but, clearly, the error is on my part.  Resolution is just to use the complete file.  This topic should remain closed.  However, one point does  remains, is that this function could be documented a little clearer in the example link.
